lib/util/git/types.ts

```diff
@@ -19,6 +19,7 @@ export interface StorageConfig {
   currentBranch?: string;
   defaultBranch?: string;
   url: string;
+  upstreamUrl?: string;
```

Comment:
The ["build-docker" test](https://github.com/renovatebot/renovate/actions/runs/13776159815/job/38525771751?pr=34532) fails with a message 
> "message": "remote: Not Found\nfatal: repository 'https://github.com/undefined.git/' not found\n"
I believe that is because the `upstreamUrl` variable is always passed in to `git.initRepo()`, even if `undefined`, but the `upstreamUrl?: string` declaration here only catches cases where the property is not passed in at all.

I believe the fix is to modify the type here.
```suggestion
  upstreamUrl?: string | undefined;
```

Not tested, thoughts? There are subtle differences between not-provided, null and undefined in TS....

Reply:
Tested, but same error. Will need to Reproduce locally and find a working fix.